lib/src/components/pagination/PaginationPages.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react'
+
+import { Stack } from '..'
+import {
+  RENDER_EIGHT_ELEMENTS,
+  TRUNCATED_THRESHOLD
+} from './pagination.constants'
+import { numOfPaginationItemsToRender } from './pagination.helper'
+import { PaginationItem } from './PaginationItem'
+import { PaginationPopover } from './PaginationPopover'
+import { usePagination } from './usePagination'
+
+const shouldTruncate = (
+  paginationItems: number[],
+  truncatedThreshold: number,
+  pagesCount: number
+): boolean => {
+  // Don't truncate when the second to last page number is visible
+  if (paginationItems?.includes(pagesCount - 1)) {
+    return false
+  } else if (pagesCount > truncatedThreshold) {
+    return true
+  }
+  return true
```

Comment:
Since you're using `return`, you don't have to use `else if`:

```suggestion
  if (paginationItems?.includes(pagesCount - 1)) return false
  if (pagesCount > truncatedThreshold) return true

  return true
```

Reply:
very good point, however I have removed this function as it turns out we didn't need its functionality in the end